### ISSUE TITLE
Kotlin String to Toast - String is not detected

### ISSUE THREAD

**Issue by monsterbrain** [xylo/intellij-postfix-templates#161]
Hi,

   I'm using Android Studio 3.6 and using the plugin I've created kotlin template like this.

```kotlin
.toast: Toast
    ANY -> android.widget.Toast.makeText(this, $expr$, android.widget.Toast.LENGTH_SHORT).show()$END$
```

However i cannot do tab-expansion on "Hello world", but can do on some variable.toast ... How to make it work ?

**Comment by xylo** [comment]
Indeed.  It works with an empty string but not with non-empty strings. :\
Have to look into it when I have some spare time.

**Comment by monsterbrain** [comment]
I'm not sure If I could be of any help, but can you please hint me so that, maybe I could have a try at fixing this ?

**Comment by xylo** [comment]
Hi @monsterbrain . Thanks for your offer to help.  I already started to look into this.  The problem is that I use(d) an expression selector that was not designed to be used for Kotlin. I just made a small test with the Kotlin specific expression selector used by the IDEA's hard coded postfix templates and that seems to be definitely the better solution. All strings get recognized with this selector. However, it detects now too many expressions. 😬 So I have to fine-tune the filter now.
I think I should also start writing automated tests so that I don't have to test each change manually (I don't want to destroy existing functionality).

**Comment by monsterbrain** [comment]
Cool. Thanks a lot. I don't know much but Kotlin can to be made even better with some postFix I guess ...

**Comment by xylo** [comment]
I tried to add automated tests for the Kotlin templates.  Unfortunately, JetBrains does not provide such a testing framework for Kotlin.  It's only available for Java.  See https://intellij-support.jetbrains.com/hc/en-us/community/posts/115000171090-Kotlin-IDEA-test-framework-library for more details.

**Comment by xylo** [comment]
Fixed in version 2.11.2.
